packages/arb-token-bridge-ui/src/hooks/useTransactionHistory.ts

```diff
@@ -0,0 +1,775 @@
+              isConnectedToParentChain
+            })
+
+            const includeReceivedTxs = shouldIncludeReceivedTxs({
+              type,
+              isSmartContractWallet,
+              isConnectedToParentChain
+            })
+            try {
+              return await fetcherFn({
+                sender: includeSentTxs ? address : undefined,
+                receiver: includeReceivedTxs ? address : undefined,
+                l1Provider: getProvider(chainPair.parentChain),
+                l2Provider: getProvider(chainPair.chain),
+                pageNumber: 0,
+                pageSize: 1000
+              })
+            } catch {
+              addFailedChainPair(prevFailedChainPairs => {
+                if (!prevFailedChainPairs) {
+                  return [chainPair]
+                }
+                if (
+                  typeof prevFailedChainPairs.find(
+                    prevPair =>
+                      prevPair.parentChain === chainPair.parentChain &&
+                      prevPair.chain === chainPair.chain
+                  ) !== 'undefined'
+                ) {
+                  // already added
+                  return prevFailedChainPairs
+                }
+
+                return [...prevFailedChainPairs, chainPair]
+              })
+
+              return []
+            }
+          })
+      )
+    },
+    [address, isTestnetMode, addFailedChainPair, isSmartContractWallet, chain]
+  )
+
+  const shouldFetch =
+    address &&
+    chain &&
+    !isLoadingAccountType &&
+    typeof isTestnetMode !== 'undefined'
+
+  const {
+    data: depositsData,
+    error: depositsError,
+    isLoading: depositsLoading
+  } = useSWRImmutable(
+    shouldFetch ? ['tx_list', 'deposits', address, isTestnetMode] : null,
+    () => fetcher('deposits')
+  )
+
+  const {
+    data: withdrawalsData,
+    error: withdrawalsError,
+    isLoading: withdrawalsLoading
+  } = useSWRImmutable(
+    shouldFetch ? ['tx_list', 'withdrawals', address, isTestnetMode] : null,
+    () => fetcher('withdrawals')
+  )
+
+  const deposits = (depositsData || []).flat()
+
+  const withdrawals = (withdrawalsData || []).flat()
+
+  // merge deposits and withdrawals and sort them by date
+  const transactions = [
+    ...deposits,
+    ...withdrawals,
+    ...combinedCctpTransfers
+  ].flat()
+
+  return {
+    data: transactions,
+    loading: depositsLoading || withdrawalsLoading || cctpLoading,
+    error: depositsError ?? withdrawalsError,
+    failedChainPairs: failedChainPairs || []
+  }
+}
+
+/**
+ * Maps additional info to previously fetches transaction history, starting with the earliest data.
+ * This is done in small batches to safely meet RPC limits.
+ */
+export const useTransactionHistory = (
+  address: `0x${string}` | undefined,
+  // TODO: look for a solution to this. It's used for now so that useEffect that handles pagination runs only a single instance.
+  { runFetcher = false } = {}
+): TransactionHistoryParams => {
```

Comment:
NIT: I find it very weird that the output of this hook is "params" (although I get it, those are basically props to the tx history component). We should rename this to `UseTransactionHistoryResult`.

And then in TransactionHistory component, define it's params/props with the type -
`type TransactionHistoryParams = UseTransactionHistoryResult & {address: ...}`

It would be much cleaner to read and understand.

Reply:
done